### PR TITLE
MembersListCard fix

### PR DESCRIPTION
**Pull request by steff-petro**
## Hey, I just made a Pull Request!

- Changed static route in MembersListCard to use a dynamic route. Solves this bug: https://github.com/backstage/backstage/issues/19289
- Updated test classes to work with dynamic route

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/org/src/components/Cards/Group/MembersList/MembersListCard.tsx [hunk 1]
```diff
@@ -105,7 +107,7 @@ const MemberComponent = (props: { member: UserEntity }) => {
               <Link
                 data-testid="user-link"
                 to={generatePath(
-                  `/catalog/:namespace/user/${metaName}`,
+                  `${catalogLink()}/:namespace/user/${metaName}`,
```
Comment (awanlin) [inline plugins/org/src/components/Cards/Group/MembersList/MembersListCard.tsx:110]: I know this is a Draft but could we have used the `EntityRefLink` component for this instead? 

https://github.com/backstage/backstage/blob/999376ee010c59edf51074f53d21a4acd4d3404d/plugins/catalog-react/src/components/EntityRefLink/EntityRefLink.tsx#L47

Comment (steff-petro) [reply plugins/org/src/components/Cards/Group/MembersList/MembersListCard.tsx:110]: That was the first thing I tried but the EntityRefLink returns the entity in this format: user:<name> (not great looking). Then once I saw that the catalogIndexRouteRef was used in another place I just went with that. But I can look into it further and see if there is a way to make it work with EntityRefLink if you think that would be more optimal.

Comment (awanlin) [reply plugins/org/src/components/Cards/Group/MembersList/MembersListCard.tsx:110]: It just seems like the perfect use for it. Have you tried with the `title` prop? Here's an example:

https://github.com/backstage/backstage/blob/999376ee010c59edf51074f53d21a4acd4d3404d/plugins/catalog/src/components/CatalogTable/columns.tsx#L54-L58

Comment (steff-petro) [reply plugins/org/src/components/Cards/Group/MembersList/MembersListCard.tsx:110]: Your right, that did it. My bad, I didn't realize the title prop replaced the value that was being displayed by the EntityRefLink. I guess that's what I get for not trying it out in the first place😅. I will do some corrections and then push the new version for review.

Comment (awanlin) [reply plugins/org/src/components/Cards/Group/MembersList/MembersListCard.tsx]: That's great, no worries, thanks for taking on the work of fixing this 👍

Comment (freben) [reply plugins/org/src/components/Cards/Group/MembersList/MembersListCard.tsx]: EntityRefLink has a defaultKind prop, that you can set to "user". That should make the prefix disappear (for users :)).

**Review by awanlin — APPROVED** [review]
This looks good to me now, thanks again @steff-petro 🚀

**Review by Rugvip — APPROVED** [review]
Nice, thank you! :pray: 

Let's :shipit:

**Comment by steff-petro** [comment]
Happy to contribute @Rugvip 😃. Thank you both for the feedback @freben @awanlin.

**Pull request merged**